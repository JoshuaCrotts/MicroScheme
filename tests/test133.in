;;; 1.3.2

(define square (λ (x) (* x x)))

((λ (x y z) (+ x y (square z))) 1 2 3)
; expect 12

(define (f x y)
  (let ((a (+ 1 (* x y)))
        (b (- 1 y)))
    (+ (* x (square a))
       (* y b)
       (* a b))))
(f 3 4)
; expect 456

(define x 5)
(+ (let ((x 3))
     (+ x (* x 10)))
   x)
; expect 38

(let ((x 3)
      (y (+ x 2)))
  (* x y))
; expect 21
