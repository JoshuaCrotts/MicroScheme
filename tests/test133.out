12
456
38
21
